### PR TITLE
Support fp8_e4m3/fp8_e5m2

### PR DESCRIPTION
**Pull request by Narsil**
# What does this PR do?

Fixes # (issue) or description of the problem this PR solves.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -99,7 +99,9 @@ Notes:
  from traditional tensor libraries perspective (torch, tensorflow, numpy, ..).
  - 0-rank Tensors (tensors with shape `[]`) are allowed, they are merely a scalar.
  - The byte buffer needs to be entirely indexed, and cannot contain holes. This prevents
- the creation of polyglot files.
```
Comment (McPatate) [inline README.md:102]: Why was this removed?

File: safetensors/src/tensor.rs [hunk 2]
```diff
@@ -670,6 +676,8 @@ impl Dtype {
             Dtype::BOOL => 1,
             Dtype::U8 => 1,
             Dtype::I8 => 1,
+            Dtype::F8_E5M2 => 1,
+            Dtype::F8_E4M3 => 1,
```
Comment (McPatate) [inline safetensors/src/tensor.rs:680]: Simple as that? 😄

**Review by McPatate — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (Narsil) [reply README.md:102]: Oops

File: safetensors/src/tensor.rs (hunk 2 above)
Comment (Narsil) [reply safetensors/src/tensor.rs:680]: Indeed :)

File: README.md [hunk 3]
```diff
@@ -99,7 +99,9 @@ Notes:
...
+ - Endianness: Little-endian.
```
Comment (Narsil) [inline README.md:102]:
```suggestion
the creation of polyglot files.
 - Endianness: Little-endian.
```

File: safetensors/src/tensor.rs (hunk 2 above)
Comment (Narsil) [reply safetensors/src/tensor.rs:680]: F4 won't be as nice :)
